modules/version-checker/version_checker.go

```diff
@@ -0,0 +1,190 @@
+package version_checker
+
+import (
+	"fmt"
+	"github.com/gruntwork-io/terratest/modules/shell"
+	"github.com/gruntwork-io/terratest/modules/testing"
+	"github.com/stretchr/testify/require"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+type VersionCheckerBinary int
+
+// List of binaries supported for version checking.
+const (
+	Docker VersionCheckerBinary = iota
+	Terraform
+	Packer
+)
+
+const (
+	// VersionRegexMatcher is a regex used to extract version string from shell command output.
+	VersionRegexMatcher = "\\d+(\\.\\d+)+"
+	// DefaultVersionArg is a default arg to pass in to get version output from shell command.
+	DefaultVersionArg = "--version"
+)
+
+type CheckVersionParams struct {
+	binary          VersionCheckerBinary
+	expectedVersion string
+	workingDir      string
+}
+
+// CheckVersionE checks whether the given binary version is greater or equal
+// to the given expected version.
+func CheckVersionE(
+	t testing.TestingT,
+	params CheckVersionParams) error {
+
+	err := validateParams(params)
+	if err != nil {
+		return err
+	}
+
+	version, err := getVersionWithShellCommand(t, params)
+	if err != nil {
+		return err
+	}
+
+	return validateBinaryVersionGreaterOrEqual(version, params.expectedVersion)
+}
+
+// CheckVersion checks whether the given binary version is greater or equal to the
+// given expected version and fail if it's not.
+func CheckVersion(
+	t testing.TestingT,
+	params CheckVersionParams) {
+	err := CheckVersionE(t, params)
+	require.NoError(t, err)
+}
+
+// Validate whether the given params contains valid data to check version.
+func validateParams(params CheckVersionParams) error {
+	// Check for empty parameters
+	if len(params.expectedVersion) == 0 {
+		return fmt.Errorf("empty expectedVersion found")
+	} else if len(params.workingDir) == 0 {
+		return fmt.Errorf("empty workingDir found")
+	} else if params.binary < 0 {
+		return fmt.Errorf("empty binary found")
+	}
+
+	// Check the format of the expected version.
+	match, err := regexp.MatchString(VersionRegexMatcher, params.expectedVersion)
+	if err != nil {
+		return fmt.Errorf(
+			"unexpected regex matcher compilation failure: %w", err)
+	} else if !match {
+		return fmt.Errorf(
+			"invalid expected version format found {%s}", params.expectedVersion)
+	}
+
+	return nil
+}
+
+// getVersionWithShellCommand get version by running a shell command.
+func getVersionWithShellCommand(t testing.TestingT, params CheckVersionParams) (string, error) {
+	// Set appropriate binary, versionArg and extract version function
+	// based on the given parameters.
+	var binaryName, versionArg = "", DefaultVersionArg
+	switch params.binary {
+	case Docker:
+		binaryName = "docker"
+	case Packer:
+		binaryName = "packer"
+	case Terraform:
+		binaryName = "terraform"
+	default:
+		t.Fatalf("unsupported binary for checking versions.")
```

Comment:
NIT: it would be great if this error message could let you know which binaries are currently supported, so the user has even less work to do to look this value up / grep through our source for it.

Reply:
SGTM. updated.